### ISSUE TITLE
[Bug] Training Loss drops to 0.0 immediately for Llama-3.2-11b-vision (works fine for Qwen2.5-VL)

### ISSUE THREAD

**Issue by cui0711** [zjysteven/lmms-finetune#87]
### Description
I am trying to fine-tune `llama-3.2-11b-vision-instruct` using this repository.
I encountered a critical issue where the **training loss drops to extremely low values or exactly 0.0** after just a few steps.

However, when I use the **exact same training pipeline and environment** to fine-tune `qwen2.5-vl-7b-instruct`, it works perfectly fine with normal loss behavior.

### Script
I used the following script to finetune Qwen:

```bash
#!/bin/bash
set -euo pipefail

export CUDA_VISIBLE_DEVICES=2,3,4,6

NUM_GPUS=4
DISTRIBUTED_ARGS="
    --nnodes=1 \
    --nproc_per_node ${NUM_GPUS} \
    --rdzv_backend c10d \
    --rdzv_endpoint localhost:0
"

# arguments that are very likely to be changed
# according to your own case
MODEL_ID=qwen2.5-vl-7b-instruct             # model id; pick on by running `python supported_models.py`
TRAIN_DATA_PATH=/dataset_path              # path to the training data json file
MODEL_LOCAL_PATH=/local_model_path
IMAGE_FOLDER=/image_folder

TRAIN_VISION_ENCODER=True                              # whether train the vision encoder
USE_VISION_LORA=True                                   # whether use lora for vision encoder (only effective when `TRAIN_VISION_ENCODER` is True)
TRAIN_VISION_PROJECTOR=True                            # whether train the vision projector (only full finetuning is supported)

USE_LORA=True                                           # whether use lora for llm
Q_LORA=False                                            # whether use q-lora for llm; only effective when `USE_LORA` is True
LORA_R=64                                                # the lora rank (both llm and vision encoder)
LORA_ALPHA=128                                            # the lora alpha (both llm and vision encoder)
LORA_DROPOUT=0.1

RUN_ID=${MODEL_ID}_lora-${USE_LORA}_xichang_pseudo    # a custom run id that determines the checkpoint folder and wandb run name

DS_STAGE=zero2                                          # deepspeed stage; < zero2 | zero3 >
PER_DEVICE_BATCH_SIZE=2                                 # batch size per GPU
GRAD_ACCUM=2                                            # gradient accumulation steps
NUM_EPOCHS=3                                            # number of training epochs
MAX_GRAD_NORM=8.0
LR=2e-5                                                 # learning rate
MODEL_MAX_LEN=8192                                       # maximum input length of the model


torchrun $DISTRIBUTED_ARGS train.py \
    --model_id $MODEL_ID \
    --model_local_path $MODEL_LOCAL_PATH \
    --data_path $TRAIN_DATA_PATH \
    --image_folder $IMAGE_FOLDER \
    --output_dir ./checkpoints/$RUN_ID \
    --report_to wandb \
    --run_name $RUN_ID \
    --deepspeed ./ds_configs/${DS_STAGE}.json \
    --bf16 True \
    --num_train_epochs $NUM_EPOCHS \
    --per_device_train_batch_size $PER_DEVICE_BATCH_SIZE \
    --per_device_eval_batch_size $PER_DEVICE_BATCH_SIZE \
    --gradient_accumulation_steps $GRAD_ACCUM \
    --max_grad_norm $MAX_GRAD_NORM \
    --eval_strategy "epoch" \
    --save_strategy "epoch" \
    --save_total_limit 2 \
    --learning_rate ${LR} \
    --weight_decay 0.01 \
    --warmup_ratio 0.1 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length $MODEL_MAX_LEN \
    --gradient_checkpointing True \
    --dataloader_num_workers 16 \
    --train_vision_encoder $TRAIN_VISION_ENCODER \
    --use_vision_lora $USE_VISION_LORA \
    --train_vision_projector $TRAIN_VISION_PROJECTOR \
    --use_lora $USE_LORA \
    --q_lora $Q_LORA \
    --lora_r $LORA_R \
    --lora_alpha $LORA_ALPHA \
    --lora_dropout $LORA_DROPOUT
```

and use this script to finetune llama-3.2-11b-vision-instruct:

```bash
#!/bin/bash
set -euo pipefail

# add two environment variables
export HF_ENDPOINT=https://hf-mirror.com
export HF_TOKEN=hf_nayp************phGDwEl

# change the model_id
MODEL_ID=llama-3.2-11b-vision-instruct 

# same as the script above
```

The only difference is that the model name has been changed and the MODEL_LOCAL_PATH parameter has been removed, while the HF_ENDPOINT environment variable has been added to download the model from https://hf-mirror.com.

Could you please confirm whether meta-llama/Llama-3.2-11B-Vision-Instruct is fully supported in this repository, and if so, let me know what additional configuration or data formatting is required to avoid the loss collapsing to 0.0?

**Comment by lavinal712** [comment]
There were some issues with the assistant tokens earlier. I’ve updated the code, so please pull the latest version. If you still encounter any problems, feel free to keep reporting them. Thank you!
